Build failure: osrm-backend

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix-build -A osrm-backend`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/300018873

### Relevant log output

```console
In function 'memcpy',
    inlined from 'move_assign' at include/boost/function/function_template.hpp:1008:24,
    inlined from 'swap' at include/boost/function/function_template.hpp:873:22,
    inlined from 'operator=.isra' at include/boost/function/function_template.hpp:1125:22:
/nix/store/x4cz3spvw0bwwz5sjsdn2qm4f89rcryn-glibc-2.40-66-dev/include/bits/string_fortified.h:29:33: warning: 'MEM <unsigned char[24]> [(char * {ref-all})&D.9070 + 8B]' is used uninitialized []
   29 |   return __builtin___memcpy_chk (__dest, __src, __len,
      |                                 ^
include/boost/function/function_template.hpp: In member function 'operator=.isra':
include/boost/function/function_template.hpp:1125:5: note: '<anonymous>' declared here
In member function 'operator()',
    inlined from 'FindSmallestEdge' at /build/source/include/util/filtered_graph.hpp:95:53,
    inlined from 'FindSmallestEdge' at /build/source/include/engine/datafacade/contiguous_internalmem_datafacade.hpp:137:46,
    inlined from 'unpackPath' at /build/source/include/engine/routing_algorithms/routing_base_ch.hpp:225:57,
    inlined from 'unpackPath' at /build/source/include/engine/routing_algorithms/routing_base_ch.hpp:380:19,
    inlined from 'unpackLegs' at /build/source/include/engine/routing_algorithms/shortest_path_impl.hpp:190:19,
    inlined from 'constructRouteResult' at /build/source/include/engine/routing_algorithms/shortest_path_impl.hpp:268:15:
/nix/store/sa7j7cddyblhcb3ch3ds10w7nw75yjj1-gcc-14.3.0/include/c++/14.3.0/bits/std_function.h:591:16: error: '<anonymous>' may be used uninitialized []
  591 |         return _M_invoker(_M_functor, std::forward<_ArgTypes>(__args)...);
      |                ^
/nix/store/sa7j7cddyblhcb3ch3ds10w7nw75yjj1-gcc-14.3.0/include/c++/14.3.0/bits/std_function.h: In function 'constructRouteResult':
/nix/store/sa7j7cddyblhcb3ch3ds10w7nw75yjj1-gcc-14.3.0/include/c++/14.3.0/bits/std_function.h:288:7: note: by argument 1 of type 'const union _Any_data & {ref-all}' to '_M_invoke' declared here
  288 |       _M_invoke(const _Any_data& __functor, _ArgTypes&&... __args)
      |       ^
/build/source/include/engine/routing_algorithms/routing_base_ch.hpp:225:57: note: '<anonymous>' declared here
  225 |         EdgeID smaller_edge_id = facade.FindSmallestEdge(
      |                                                         ^
lto1: all warnings being treated as errors
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.34, NixOS, 25.11 (Xantusia), 25.11.20250624.4b1164c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/gi1bdb5fzaqn0qhhhqwr3v46p6zndvsh-source`

### Notify maintainers

@erictapen

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/os/osrm-backend/package.nix
+++ b/pkgs/by-name/os/osrm-backend/package.nix
@@ -45,8 +45,12 @@ stdenv.mkDerivation rec {
     expat
   ];
 
-  # Needed with GCC 12
-  env.NIX_CFLAGS_COMPILE = "-Wno-error=uninitialized";
+  env.NIX_CFLAGS_COMPILE = toString [
+    # Needed with GCC 12
+    "-Wno-error=uninitialized"
+    # Needed with GCC 14
+    "-Wno-error=maybe-uninitialized"
+  ];
 
   postInstall = ''
     mkdir -p $out/share/osrm-backend

